chef/cookbooks/network/recipes/default.rb

```diff
@@ -314,9 +334,76 @@ def kill_nic(nic)
   if network.add_ovs_bridge
     bridge = network.bridge_name || "br-#{network.name}"
 
+    # Check if our current base interface (bond, vlan or physical) is already
+    # attached to another OVS bridge. Take care about that appropriately.
+    if our_iface.ovs_master && our_iface.ovs_master.name != bridge
+      old_ovs = our_iface.ovs_master
+
+      # We are already attached to an ovs bridge, check whether that attachment
+      # was already verified (or created) during this chef-client run, by looking
+      # through the list of interfaces that we already touched ('ifs'). If it is,
+      # find out which crowbar network requires that attachement and error out
+      # with a meaningful error message.
+      if ifs.key? old_ovs.name
+        other_net = sorted_networks.find do |n|
+          n.bridge_name == old_ovs.name
+        end
+        other_net ||= sorted_networks.find do |n|
+          n.name == old_ovs.name.sub(/\A#{Regexp.escape("br-")}/, "")
+        end
+        raise "The current conduit mapping for network '#{network.name}' requires " \
+          "interface '#{our_iface.name}' to be attached to ovs-brige '#{bridge}'. " \
+          "But network '#{other_net.name}' which was already set up, requires " \
+          "the same interface to be attached to ovs-bridge '#{old_ovs.name}'. " \
+          "Refusing to reconfigure. There seems to be a conflict in the conduit" \
+          "mappings for the networks '#{other_net.name}' and '#{network.name}'."
+      end
+
+      # We didn't create the attachement during the current run of chef-client.
+      # Tear down the attachement (and bridge configuration) to be able to attach
+      # the current interface to the new bridge.
+      Chef::Log.warn("Current conduit mapping for network '#{network.name}' requires " \
```

Comment:
Not sure whether I like this. This would take the other bridge down more or less silently (with just a log message), wouldn't it, potentially breaking functionality that was there, previously. Shouldn't we fail barclamp application and require the user to indicate that the action just performed was indeed fine (by deleting a state file or something like that) manually?

Reply:
@jgrassler Actually the block above this (line 347 to 360) is trying to make sure that this change is intentional an won't break anything. If the old bridge (the one we're tearing down here) is needed by any of the  enabled crowbar networks on this node the chef run will fail with a the pretty verbose usable error message above, latest when trying to recreate that bridge (the timing of that depends on the order in which the networks are processed, but it will fail at some point).